scipy/stats/tests/test_stats.py

```diff
@@ -4205,6 +4205,149 @@ def test_ttest_ind_with_uneq_var():
+        assert_allclose(res.statistic, -4.246117, atol=1e-6)
+
+    def test_compare_r_2(self):
+        '''
+        see: https://rdrr.io/cran/PairedData/man/yuen.t.test.html
+        library(PairedData)
+
+        > a <-c(-0.84504783, 0.13366078, 3.53601757, -0.62908581, 0.54119466,
+        +       -1.16511574, -0.08836614, 1.18495416, 2.48028757, -1.58925028,
+        +       -1.6706357, 0.3090472, -2.12258305, 0.3697304, -1.0415207,
+        +       -0.57783497, -0.90997008, 1.09850192, 0.41270579, -1.4927376)
+        > b <-c(1.2725522, 1.1657899, 2.7509041, 1.2389013, -0.9490494,
+        +       -1.0752459, 1.1038576, 2.9912821, 3.5349111, 0.4171922,
+        +       1.0168959, -0.7625041, -0.4300008, 3.0431921, 1.6035947,
+        +       0.5285634, -0.7649405, 1.5575896, 1.3670797, 1.1726023)
+        > yuen.t.test(a, b)
+
+            Two-sample Yuen test, trim=0.2
+
+        data:  a and b
+        t = -3.0983, df = 21.749, p-value = 0.005293
+        alternative hypothesis: true difference in trimmed means is not equal \
+        to 0
+        95 percent confidence interval:
+         -2.157551 -0.426652
+        sample estimates:
+        trimmed mean of x trimmed mean of y
+               -0.2908835         1.0012182
+
+        '''
+        a = [-0.84504783, 0.13366078, 3.53601757, -0.62908581, 0.54119466,
+             -1.16511574, -0.08836614, 1.18495416, 2.48028757, -1.58925028,
+             -1.6706357, 0.3090472, -2.12258305, 0.3697304, -1.0415207,
+             -0.57783497, -0.90997008, 1.09850192, 0.41270579, -1.4927376]
+        b = [1.2725522, 1.1657899, 2.7509041, 1.2389013, -0.9490494,
+             -1.0752459, 1.1038576, 2.9912821, 3.5349111, 0.4171922,
+             1.0168959, -0.7625041, -0.4300008, 3.0431921, 1.6035947,
+             0.5285634, -0.7649405, 1.5575896, 1.3670797, 1.1726023]
+        res = stats.ttest_ind(a, b, trim=20, equal_var=False)
+        assert_allclose(res.pvalue, 0.005293306, atol=1e-7)
+        assert_allclose(res.statistic, -3.098332, atol=1e-6)
+
+    def test_compare_SAS(self):
+        # Source of the data used in this test:
+        # https://support.sas.com/resources/papers/proceedings14/1660-2014.pdf
+        a = [12, 14, 18, 25, 32, 44, 12, 14, 18, 25, 32, 44]
+        b = [17, 22, 14, 12, 30, 29, 19, 17, 22, 14, 12, 30, 29, 19]
+        # In this paper, a trimming percentage of 5% is used. However,
+        # in their implementation, the number of values trimmed is rounded to
+        # the nearest whole number. However, consistent with
+        # `scipy.stats.trimmed_mean`, this test truncates to the lower
+        # whole number. In this example, the paper notes that 1 value is
+        # trimmed off of each side. 10% replicates this amount of trimming.
+        res = stats.ttest_ind(a, b, trim=9, equal_var=False)
+        assert_allclose(res.pvalue, 0.514522, atol=1e-6)
+        assert_allclose(res.statistic, 0.669169, atol=1e-6)
+
+    def test_axis(self):
```

Comment:
This is the right idea, but I think `ttest_ind` works with any number of dimensions, and also w/ broadcasting (e.g. in this example, `b` could be of size `(8, 1)` and each column of `a` would get tested against that single column of `b`), so we should test a little more generally. 

See if you can adapt `test_ttest_ind_permutations_many_dims` to work as a test for the trimmed mean tests. If it doesn't work, it might be easier if you [broadcast](https://numpy.org/doc/stable/user/basics.broadcasting.html) the input arrays instead of relying on all the functions you're using to do it properly. I don't know if `[np.broadcast_arrays](https://numpy.org/doc/stable/reference/generated/numpy.broadcast_arrays.html)` will do what you need it to do because your arrays are allowed to have different lengths along `axis`; it won't allow that. So you might want to take a look at `_broadcast_concatenate` in `stats.py` - that approach might help you broadcast properly while ignoring `axis`. I think it you need to do the same thing, except without concatenating the results.

Reply:
I changed the calls in `test_ttest_ind_permutations_many_dims` to `ttest_ind` to use trimming instead of permutations, and it appears to work. Looking at `test_ttest_ind_permutations_many_dims` briefly, it seems that it checks the vectorized result against each slice already -- is there more to change?

Reply:
That's great. Nothing more to do. Maybe parametrize that test rather than copy it then.

Reply:
Parametrize is ideal since it is just a matter of setting the `kwds`. I can pull that test out of its current class (`Test_ttest_ind_permutations`) and make it its own function, `test_ttest_many_dims`. I don't see another class that testing both permutations and trimming cleanly fits but we could make one.

Reply:
If there's a class for ttest_ind overall, put it there. If not, yeah, create one.